### PR TITLE
Improve type annotation for image layer and related

### PR DESCRIPTION
**Pull request by Czaki**
Result of random walk over code when reading it. Add few type annotations

### CODE REVIEW

File: napari/layers/image/_image_constants.py [hunk 1]
```diff
@@ -1,9 +1,29 @@
...
+    "hanning",
+    "hermite",
+    "kaiser",
+    "lanczos",
+    "mitchell",
+    "nearest",
+    "spline16",
+    "spline36",
+    "custom",
+]
```
Comment (brisvag) [inline napari/layers/image/_image_constants.py:25]: this feels so redundant though... Can't `StringEnum` somehow already capture this?

Comment (Czaki) [reply napari/layers/image/_image_constants.py:25]: We need python 3.11 support of `*` unpack in `[]` to make it generic. 

Maybe some stub generation could be a good workaround.

**Comment by Carreau** [comment]
Conflict resolved, and merged main into this.

**Comment by Carreau** [comment]
Conflict resolved, and merged main into this a second time.

**Comment by Carreau** [comment]
Ok, this picked up some types errors with the changes in main

File: napari/layers/image/_image_constants.py (hunk 1 above)
Comment (Carreau) [reply napari/layers/image/_image_constants.py:25]: I had to add some casts here and there for typing to be happy. I think we can just make an issue once this is merged to refactor that once we are 3.11+ only.

**Review by Carreau — APPROVED** [review]

**Pull request merged**